common/state/state_kv.cpp

```diff
@@ -71,6 +71,9 @@ pdo::state::State_KV::State_KV(const StateBlockId& id, const ByteArray& key)
 
 void pdo::state::State_KV::Finalize(ByteArray& outId)
 {
+    pdo::error::ThrowIf<pdo::error::RuntimeError>(
+        kv_start_mode == KV_UNINITIALIZED, "attempt to finalize uninitialized state");
```

Comment:
doesn't this error really mean that there was a problem during unpack? and if that's the case, does throwing yet another exception actually help? if it is uninitialized here then we don't need to clean anything up. maybe just log and move on with the rest of the clean up.

Reply:
before, yes, this was symptom of a possible error during unpack, and the exception was thrown in ContractState::Finalize. But now:
1) you cannot check anymore for a null pointer
2) finalize is not called in the catch block
3) you are doing something wrong calling finalize on the uninitialized state (I think this is the main reason)
4) arguably unpack is less likely to fail; rather it fails the constructor of ContractState, called in contract_enclave.cpp after the request is constructed

Reply:
still not sure i understand. the errors will have already been thrown. i can see this as an assert. but all meaningful errors should have resulted in throws long before we get here.